server/src/main/java/org/opensearch/gateway/AsyncShardFetch.java

```diff
@@ -130,11 +151,27 @@ public synchronized int getNumberOfInFlightFetches() {
      * The ignoreNodes are nodes that are supposed to be ignored for this round, since fetching is async, we need
      * to keep them around and make sure we add them back when all the responses are fetched and returned.
      */
-    public synchronized FetchResult<T> fetchData(DiscoveryNodes nodes, Set<String> ignoreNodes) {
+    public synchronized FetchResult<T> fetchData(DiscoveryNodes nodes, Map<ShardId, Set<String>> ignoreNodes) {
```

Comment:
In line with the other comment of moving shard attrs in a separate class, we can probably do the same here?